src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -116,8 +116,11 @@ PeerImp::run()
         if ((hello_.has_ledgerclosed ()) && (
             hello_.ledgerclosed ().size () == (256 / 8)))
         {
-            memcpy (closedLedgerHash_.begin (),
-                hello_.ledgerclosed ().data (), 256 / 8);
+            {
+                std::lock_guard<std::mutex> sl(recentLock_);
+                memcpy (closedLedgerHash_.begin (),
+                    hello_.ledgerclosed ().data (), 256 / 8);
```

Comment:
I know this isn't anything new you added, but while in this neighborhood of code, do you mind giving a good identifier to the expression `256/8`? It is now used four times within this method and seventeen times within this file.

Reply:
`closedLedgerHash_` is a `uint256` and `hello_.ledgerclosed()` is a `std::string` that contains the raw bytes for the `uint256`. The `256 / 8` is the size, in bytes, of a 256-bit variable.

Silly, I know. It's poor form and bad style in code that hasn't been rewritten cleanly.
 
Personally, I'd prefer the `if` on line 118 to use:

`m->ledgerhash ().size () == closedLedgerHash_.size()`

We could add a second overload of `copyFrom` that would do the `memcpy` (and the size check) allowing us to change this code to:

`closedLedgerHash_.copyFrom(m->ledgerhash().begin(), m->ledgerhash().end());`

Alternatively, we could just avoid adding a function and using code that's simply more readable:

`std::memcpy(closedLedgerHash_.begin(), hello_.ledgerclosed().data(), closedLedgerHash_.size());`

or even:

`std::copy(m->ledgerhash().begin(), m->ledgerhash().end(), closedLedgerHash_.begin());`

Reply:
@nbougalis, I like your suggestion of replacing the magic numbers with `closedLedgerHash_.size()`.

I'm inclined to take a slightly different tack and use `sizeof (closedLedgerHash_)`.  To me that has the tiny advantage of clearly (to the reader) being a compile-time constant.  `base_uint::size()` is `constexpr`, so it is probably handled as a compile-time constant, at least by Clang.  But, as a reader, you have to go the the declaration of `base_uint::size()` to find that out.

Regardless, I think the outcome is comparable.  Replace a magic number with a value that is clearly associated with the correct value/type.

Reply:
@scottschurr, I like that idea but it's dangerous. We need to be _very_ careful with `sizeof(variable)` in this context because that hinges on assumptions that, while true today, may not be true tomorrow. In this case, it assumes that the size of a `base_uint<N>` is the same as `N / 8`.; that's not guaranteed.

Of course, this is something we rely on anyways, so it's not a stretch, but you can see how it could lead to trouble if someone refactors the code. It may be a good idea to add a `static_assert(std::is_pod(base_uint<...>), "Ack, base_uint needs to be POD")` or `std::is_trivial` or what-have-you, so that if things change out from under us, at least, we'll know right away instead of continuing oblivious to the change.

Reply:
@nbougalis, good point.  Yes, I'll add that `static_assert`.